### ISSUE TITLE
Disable link previews

### ISSUE THREAD

**Issue by kenniki** [vysheng/tg#502]
Yo, thanks for this project. I massively enjoy it.
I wish I could give you money.

Anyway, is there a way for tg-cli to do the equivalent of clicking the X on this new link preview stuff?
It's a great feature in theory but if you're running a bot with it it's just annoying beyond belief and the Desktop clients don't have a way of disabling it.

I hooked up Telegram to IFTTT and this is what it looks like now:

![screenshot_20150430-211042](https://cloud.githubusercontent.com/assets/6758700/7420535/9d8fe0c6-ef7d-11e4-8a74-a747b4e6481b.png)

**Comment by LukeLR** [comment]
The link previews are generated on the client side, therefore you can't disable link previews for messages sent by telegram cli. You probably should request the feature of disabling link previews at the developers of the telegram GUI clients

**Comment by kenniki** [comment]
But that's wrong.
If I paste a link into the message box, Telegram offers me a link preview. I can hit enter to send it or click the little X to not send it. If I click the little X, everyone who receives the message doesn't see the link preview either. This is not a client problem (although a way to disable that there, too, would be neat).
The feature can be opted out of from the senders side. I'd like to see tg-cli having an implementation of that.

**Comment by vysheng** [comment]
added key (--disable-link-preview) to disable previews for outgoing messages

**Comment by kenniki** [comment]
vysheng, you are a god.
Thanks so much.
I still want to give you money.

**Comment by luckydonald** [comment]
Is this globally defined for the CLI,
 or is it possible to turn previews off on per-message basis (at least in Lua)?

**Comment by phoudoin** [comment]
I guess that should not be difficult to add a flag to tgl_message and check both global and message flags in tgl_do_send_msg()

**Comment by afanjul** [comment]
Yes please, its very annoying the link preview feature for everything its not a media link (like videos or audios). An option to enable/disable "Media Preview" and "Links previews" should be availabe.

**Comment by luckydonald** [comment]
@vysheng will this be implemented on a per-message basis? (e.g. lua?)

**Comment by luckydonald** [comment]
Another solution would be a command to enable or disable it in the running cli

**Comment by vysheng** [comment]
probably it works prepend command with ```[disable_preview]``` i.e. ```[disable_preview] msg chat https://github.com/vysheng/tg```

**Comment by luckydonald** [comment]
Worked here.

**Comment by phoudoin** [comment]
Please make it exposed as a flag in tgl. In TG it make sense to have some statefull switch, but at library level this should be per message if needed.

**Comment by luckydonald** [comment]
He has a point there.

**Comment by animalillo** [comment]
Thanks, it helped a lot!
